vault/core.go

```diff
@@ -713,6 +713,9 @@ type CoreConfig struct {
 
 	Logger log.Logger
 
+	// Use the deadlocks library to detect deadlocks
+	DetectDeadlocks string
```

Comment:
Previously, with the build tag based approach, we were able to have the detection on by default for our CI tests.  Can we maintain that behaviour, at least for NewTestCluster?  I'm thinking, since we want it on by default for NewTestCluster, that maybe it merits a new field in TestClusterOptions.  That's a little bit weird, since it's a CoreConfig field, and there's already a CoreConfig argument to NewTestCluster.  But by convention I'd expect that CoreConfig field to default off, whereas I want the NewTestCluster behaviour to default on.

Further complications: NewTestCluster is also used in a couple of user-facing CLI commands, -dev-three-node and -dev-four-cluster, and in *those* cases deadlock detection should be off.

Finally, can you remove references to the `deadlock` build tag in files under `.circleci`?  Oh wait, I just looked at them, and I'm seeing that we only enable deadlock detection for the non `-race` tests.  Hmm.  Maybe instead of a new TestClusterOption field, NewTestCluster should be controlled by the build tag, which we keep around for that purpose?  The issue is that running tests with -race can consume an order of magnitude more CPU/RAM, and that can trigger false positives in the deadlock detection.

Reply:
got it! I added this constant that changes based on build tag https://github.com/hashicorp/vault/pull/18604/files#diff-07bbf23b142e3fba2ba348837bc6814262f0cabd708bcc3fb9ff52a91ef079d3R1371.